### ISSUE TITLE
Unable to see "active" job in connector interface when there are more than 50 jobs in progress.

### ISSUE THREAD

**Issue by romain-filigran** [LuatixHQ/opencti#8015]
## Description

When a connector has more than 50 jobs "in Progress", we can't have a vision of the "active" job as the page only list the 50 most recent jobs. Related connectors: Crowdstrike.

As a first step, it might be a good idea to add a counter for the number of jobs in progress.
Then see if you can paginate, filter or display more than 50 items.
@richard-julien @Kedae @labo-flg : Any suggestions ?

Maybe we should also take a look at the Crowdstrike connector to understand why it's generating so many jobs.

OpenCTI version: 6.2.13

## Reproducible Steps

https://www.notion.so/filigran/All-Connectors-1709057d28914ed293ea867e9cc2e18b

**Comment by labo-flg** [comment]
API side, pagination is already implemented so we could migrate these 2 panels (in progress and completed works) into paginated tables.

**Comment by romain-filigran** [comment]
Reviewed today. Corrective actions to be implemented: 
1. Add a counter for the number of jobs in progress.
2. Change the sort order and sort by most oldest in order to see "Active" In progress jobs first.

**Comment by CelineSebe** [comment]
After a discussion with @helene-nguyen, we reported another issue on this screen. When I reset the connector status, as you can see below, the job is not really completed, yet it appears under completed works:

![Image](https://github.com/user-attachments/assets/f2af5d3a-bc0b-4518-a9c7-3a9835f9b180)

**Comment by labo-flg** [comment]
@aHenryJard might have some insights for you :)

What I understood is that the completion status gets to "completed" once the number of completed operations is equal to the count of expectations.
Now with connectors that work in streaming (like the new behavior of the CSV feed), the number of expectations might grow, as long as there is more to process, batch after batch.
Depending on the resources and processing, the status might turn "completed" and then expectation would grow.... but status remain "completed".
Something to look at and improve probably.

**Comment by aHenryJard** [comment]
The issue for COMPLETE status that never goes back to in progress when there is additional work that is recorded, is that the status is not moved back to in progress in `updateExpectationsNumber`:

 https://github.com/OpenCTI-Platform/opencti/blob/52666810475b4163018bb8c40701335d0f1efc07/opencti-platform/opencti-graphql/src/domain/work.js#L239-L246

**Comment by aHenryJard** [comment]
Note that there is also a mechanism of moving "old" job as complete, which means only one work is in progress at a time

https://github.com/OpenCTI-Platform/opencti/blob/52666810475b4163018bb8c40701335d0f1efc07/opencti-platform/opencti-graphql/src/manager/connectorManager.js#L27

**Comment by richard-julien** [comment]
Please dont mix differnet problem in the same ticket.
This one is for pagination and ordering.
Lets discuss about the completion and how the system work on another ticket

**Comment by aHenryJard** [comment]
yes I agree @richard-julien ; but then having 50 work in progress is not possible anymore. So we can also pause this issue while discussing on in progress/complete work behavior.

**Comment by richard-julien** [comment]
For me you can have more than 50 work not started, that will produce the same issue of pagination
